adafruit_mcp2515/__init__.py

```diff
@@ -164,28 +164,57 @@
     ["CTRL_REG", "STD_ID_REG", "INT_FLAG_MASK", "LOAD_CMD", "SEND_CMD"],
 )
 
-# This is magic, don't disturb the dragon
-# expects a 16Mhz crystal
 _BAUD_RATES = {
-    # CNF1, CNF2, CNF3
-    1000000: (0x00, 0xD0, 0x82),
-    500000: (0x00, 0xF0, 0x86),
-    250000: (0x41, 0xF1, 0x85),
-    200000: (0x01, 0xFA, 0x87),
-    125000: (0x03, 0xF0, 0x86),
-    100000: (0x03, 0xFA, 0x87),
-    95000: (0x03, 0xAD, 0x07),
-    83300: (0x03, 0xBE, 0x07),
-    80000: (0x03, 0xFF, 0x87),
-    50000: (0x07, 0xFA, 0x87),
-    40000: (0x07, 0xFF, 0x87),
-    33000: (0x09, 0xBE, 0x07),
-    31250: (0x0F, 0xF1, 0x85),
-    25000: (0x0F, 0xBA, 0x07),
-    20000: (0x0F, 0xFF, 0x87),
-    10000: (0x1F, 0xFF, 0x87),
-    5000: (0x3F, 0xFF, 0x87),
-    666000: (0x00, 0xA0, 0x04),
+    # This is magic, don't disturb the dragon
+    # expects a 16Mhz crystal
+    16000000: {
+        # CNF1, CNF2, CNF3
+        1000000: (0x00, 0xD0, 0x82),
+        500000: (0x00, 0xF0, 0x86),
+        250000: (0x41, 0xF1, 0x85),
+        200000: (0x01, 0xFA, 0x87),
+        125000: (0x03, 0xF0, 0x86),
+        100000: (0x03, 0xFA, 0x87),
+        95000: (0x03, 0xAD, 0x07),
+        83300: (0x03, 0xBE, 0x07),
+        80000: (0x03, 0xFF, 0x87),
+        50000: (0x07, 0xFA, 0x87),
+        40000: (0x07, 0xFF, 0x87),
+        33000: (0x09, 0xBE, 0x07),
+        31250: (0x0F, 0xF1, 0x85),
+        25000: (0x0F, 0xBA, 0x07),
+        20000: (0x0F, 0xFF, 0x87),
+        10000: (0x1F, 0xFF, 0x87),
+        5000: (0x3F, 0xFF, 0x87),
+        666000: (0x00, 0xA0, 0x04),
+    },
+    # Values based on this calculator, for 8MHz, controller MCP2510:
+    # https://www.kvaser.com/support/calculators/bit-timing-calculator/
+    8000000: {
+        # CNF1, CNF2, CNF3
+        1000000: (
+            0x00,
+            0x91,
+            0x01,
+        ),  # seems it is not possible, this values may be wrong
```

Comment:
What do you mean by "these values may be wrong"?

Reply:
Using the calculator on https://www.kvaser.com/support/calculators/bit-timing-calculator/, seems it is not possible to use the 1000000 with a 8MHz crystal.

Just did a commit with the values / option removed. I think is the correct thing to do.

Reply:
Solved on last PR. I just removed that option.